src/Modules/OrchardCore.Commerce/Services/ShoppingCartHelpers.cs

```diff
@@ -83,7 +83,10 @@ public async Task<ShoppingCartItem> ParseCartLineAsync(ShoppingCartLineUpdateMod
         {
             await _notifier.AddAsync(
                 NotifyType.Error,
-                T["Can't add product {0} because it doesn't have a price.", line.ProductSku]);
+                // Localized strings need to be in one line.
+#pragma warning disable S103 // Long line (which is greater than 150 authorized)
+                T["Can't add product {0} because it doesn't have a price, or its currency doesn't match with the current display currency.", line.ProductSku]);
```

Comment:
I know I'm too late for the party, but `match` sounds better without `with`:

```suggestion
                T["Can't add product {0} because it doesn't have a price, or its currency doesn't match the current display currency.", line.ProductSku]);
```

Same in the view.